### ISSUE TITLE
Test failure System.IO.Tests.Directory_EnumFSE_str_str_so.PatternTests_DosQM(pattern: \"f???.txt\", sourceFiles: [\"f.txt\", \"foo.txt\", \"foob.txt\", \"fooba.txt\", \"foobar.txt\"], expected: [\"f.txt\", \"foo.txt\", \"foob.txt\"])

### ISSUE THREAD

**Issue by VincentBu** [dotnet/runtime#54304]
Run: [runtime 20210616.88](https://dev.azure.com/dnceng/public/_build/results?buildId=1190871&view=ms.vss-test-web.build-test-results-tab&runId=35753054&resultId=138753&paneView=debug)

Failed test:
```
net6.0-windows-Release-x86-CoreCLR_release-Windows.7.Amd64.Open

- System.IO.Tests.Directory_EnumFSE_str_str_so.PatternTests_DosQM(pattern: \"f???.txt\", sourceFiles: [\"f.txt\", \"foo.txt\", \"foob.txt\", \"fooba.txt\", \"foobar.txt\"], expected: [\"f.txt\", \"foo.txt\", \"foob.txt\"])

```
**Error message:**
```
System.IO.IOException : The handle is invalid. : 'C:\h\w\A7C2092A\t\Directory_EnumFSE_str_str_so_q4ywmcyq.ln3\PrepareDirectory_539_4ee05970\foob.txt'


Stack trace
   at System.IO.RandomAccess.WriteAtOffset(SafeFileHandle handle, ReadOnlySpan`1 buffer, Int64 fileOffset, String path) in /_/src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.Windows.cs:line 77
   at System.IO.Strategies.SyncWindowsFileStreamStrategy.WriteSpan(ReadOnlySpan`1 source) in /_/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/SyncWindowsFileStreamStrategy.cs:line 104
   at System.IO.Strategies.BufferedFileStreamStrategy.Flush(Boolean flushToDisk) in /_/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/BufferedFileStreamStrategy.cs:line 792
   at System.IO.FileStream.Flush() in /_/src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs:line 350
   at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder) in /_/src/libraries/System.Private.CoreLib/src/System/IO/StreamWriter.cs:line 327
   at System.IO.StreamWriter.Dispose(Boolean disposing) in /_/src/libraries/System.Private.CoreLib/src/System/IO/StreamWriter.cs:line 220
   at System.IO.File.WriteAllText(String path, String contents) in /_/src/libraries/System.Private.CoreLib/src/System/IO/File.cs:line 307
   at System.IO.Tests.Directory_GetFileSystemEntries_str_str.PrepareDirectory(String[] sourceFiles, Boolean useExtendedPaths) in /_/src/libraries/System.IO.FileSystem/tests/Directory/GetFileSystemEntries_str_str.cs:line 539
   at System.IO.Tests.Directory_GetFileSystemEntries_str_str.PatternTests_DosQM(String pattern, String[] sourceFiles, String[] expected) in /_/src/libraries/System.IO.FileSystem/tests/Directory/GetFileSystemEntries_str_str.cs:line 242
```

**Comment by danmoseley** [comment]
@dotnet/area-system-io related to FileStream work?

**Comment by adamsitnik** [comment]
Fixed by @danmoseley in #54660 (this is the first time we see it on Windows btw)
